### DISCUSSION TITLE
MultiInputNet - Create prediction on new test set

### DISCUSSION THREAD

**Discussion by shayp** [timeseriesAI/tsai#345] [Q&A]
I'm having difficulties running prediction on a new test data set on MultiInputNet (TabModel and tsmodel).
Iv'e Created dl for each input and combined it into MixedDataLoader.

`
test_dl_tab = tab_dls.test_dl(X_features_test)
test_dl_sig = ts_dls.new_dl(X=X_signal_test, y=y_test)
mixed_test_dls = [test_dl_sig, test_dl_tab]
 
device = mixed_dls.train.device
mixed_test_dl = MixedDataLoader(*mixed_test_dls, shuffle=False)
test_probas, test_targets, test_preds = learn.get_preds(dl=mixed_test_dl, with_decoded=True,save_preds=None,save_targs=None,with_loss=False)
`

But when running learn.get_preds I get 
TypeError: cross_entropy_loss(): argument 'target' (position 2) must be Tensor, not tuple

**Comment by strakehyr** [comment]
Hi @shayp, I am having the same issue and posted it on #360.

**Reply by shayp** [reply]
Hi,
After trying to make this work In several ways I decided to skip the high level API of get_preds and instead call directly to learner.model. 
working directly with the model was the easiest workaround.

**Reply by ThomasBourgeois** [reply]
Hi how did you do that ? 
On my side I'm creating a mixed dls of the test : 

```
dls_ts = get_ts_dls(X_ts_test, None, splits=None,...  bs=bs)

dls_tab = get_tabular_dls(X_tab_test, ...  y_names=None, splits=None, bs=bs)

dls = get_mixed_dls(dls_ts, dls_tab, bs=bs)
```

Then when tyring to predict : 
```learn.model(dls.one_batch())```

I'm having errors on the size : 
```TypeError: forward() takes 2 positional arguments but 3 were given```
and can't seem to get a prediction this way.
